File: uwp/IotCoreGatewayCpp/SetOfCppModules/Module2.cpp
#include "pch.h"
#include "Module2.h"

using namespace SetOfCppModules;
using namespace Platform;

Module2::Module2()
{
}

void SetOfCppModules::Module2::Create(Microsoft::Azure::IoT::Gateway::MessageBus ^bus, Platform::String ^configuration)
{
	std::wstring formattedText = L"SetOfCppModules::Module2::Create\r\n";
	OutputDebugString(formattedText.c_str());
}

void SetOfCppModules::Module2::Destroy()
{
	std::wstring formattedText = L"SetOfCppModules::Module2::Destroy\r\n";
	OutputDebugString(formattedText.c_str());
}

void SetOfCppModules::Module2::Receive(Microsoft::Azure::IoT::Gateway::IGatewayModule ^source, Microsoft::Azure::IoT::Gateway::Message ^received_message)
{
	std::wstring content = received_message->GetContent()->Data();
	auto props = received_message->GetProperties();

	std::wstring sb = L"SetOfCppModules.Module2.Receive: Content=";
	sb += content;
	sb += L"\r\n";

	sb += L"SetOfCppModules.Module2.Receive: Properties={";

	std::for_each(begin(props), end(props), [&](Windows::Foundation::Collections::IKeyValuePair<String^, String^>^ item)
	{
		sb += item->Key->Data();
		sb += L"=";
		sb += item->Value->ToString()->Data();
		sb += L", ";
	});
	sb += L"}\r\n";

	OutputDebugString(sb.c_str());
}

File: uwp/IotCoreGatewayCpp/SetOfCsModules/Module2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Azure.IoT.Gateway;

namespace SetOfCsModules
{
    public sealed class Module2 : Microsoft.Azure.IoT.Gateway.IGatewayModule
    {
        public void Create(MessageBus bus, string configuration)
        {
            System.Diagnostics.Debug.WriteLine("SetOfCsModules.Module2.Create");
        }

        public void Destroy()
        {
            System.Diagnostics.Debug.WriteLine("SetOfCsModules.Module2.Destroy");
        }

        public void Receive(Message received_message)
        {
            string content = received_message.GetContent();
            var props = received_message.GetProperties();

            StringBuilder sb = new StringBuilder();
            sb.Append("SetOfCsModules.Module2.Receive: Content=" + content + "\r\n");
            sb.Append("SetOfCsModules.Module2.Receive: Properties={");
            foreach (var key in props.Keys)
            {
                sb.Append(key);
                sb.Append("=");
                sb.Append(props[key] as string);
                sb.Append(", ");
            }
            sb.Append("}\r\n");

            System.Diagnostics.Debug.WriteLine(sb.ToString());
        }
    }
}
